migrate.py

```diff
@@ -678,6 +723,321 @@ def copy_tests(plugin, coll, spec, args):
+                    _rewrite_yaml_mapping(el[key][idx], namespace, collection, spec)
+                else:
+                    if key == 'module_blacklist':
+                        for coll in spec.keys():
+                            if item in get_plugins_from_collection(coll, 'modules', spec):
+                                el[key][idx] = get_plugin_fqcn(namespace, coll, el[key][idx])
+                                integration_tests_add_to_deps(collection, coll)
+                    if isinstance(el[key][idx], str):
+                        # FIXME move to a func
+                        el[key][idx] = _rewrite_yaml_lookup(el[key][idx], namespace, collection, spec)
+                        el[key][idx] = _rewrite_yaml_filter(el[key][idx], namespace, collection, spec)
+                        el[key][idx] = _rewrite_yaml_test(el[key][idx], namespace, collection, spec)
+        elif isinstance(value, str):
+            el[key] = _rewrite_yaml_lookup(el[key], namespace, collection, spec)
+            el[key] = _rewrite_yaml_filter(el[key], namespace, collection, spec)
+            el[key] = _rewrite_yaml_test(el[key], namespace, collection, spec)
+
+
+def _rewrite_yaml_lookup(value, namespace, collection, spec):
+    if not ('lookup(' in value or 'query(' in value or 'q(' in value):
+        return value
+
+    for coll in spec.keys():
+        for plugin_name in get_plugins_from_collection(coll, 'lookup', spec):
+            if plugin_name not in value:
+                continue
+            value = value.replace(plugin_name, get_plugin_fqcn(namespace, coll, plugin_name))
+            integration_tests_add_to_deps(collection, coll)
+
+    return value
+
+
+def _rewrite_yaml_filter(value, namespace, collection, spec):
+    if '|' not in value:
+        return value
+
+    for coll in spec.keys():
+        for filter_plugin_name in get_plugins_from_collection(coll, 'filter', spec):
+            imported_module = import_module('ansible.plugins.filter.' + filter_plugin_name)
+            fm = getattr(imported_module, 'FilterModule', None)
+            # FIXME import once
+            if fm is None:
+                continue
+            filters = fm().filters().keys()
+            for found_filter in (match[5] for match in FILTER_RE.findall(value)):
+                if found_filter not in filters:
+                    continue
+                value = value.replace(found_filter, get_plugin_fqcn(namespace, coll, found_filter))
+                integration_tests_add_to_deps(collection, coll)
+
+    return value
+
+
+def _rewrite_yaml_test(value, namespace, collection, spec):
+    if ' is ' not in value:
+        return value
+
+    for coll in spec.keys():
+        for test_plugin_name in get_plugins_from_collection(coll, 'test', spec):
+            imported_module = import_module('ansible.plugins.test.' + test_plugin_name)
+            tm = getattr(imported_module, 'TestModule', None)
+            # FIXME import once
```

Comment:
It doesn't really matter: subsequent imports hit the cache and return already loaded modules.